cmd/fs-v1-multipart.go

```diff
@@ -830,7 +842,8 @@ func (fs fsObjects) CompleteMultipartUpload(bucket string, object string, upload
 	}
 
 	// Remove entry from `uploads.json`.
-	if err = fs.removeUploadID(bucket, object, uploadID, rwlk); err != nil {
+	removeObjectDir, err= fs.removeUploadID(bucket, object, uploadID, rwlk)
```

Comment:
As `removeObjectDir` is set only here, you could move out of usage from `defer` and have the logic here.  You could check below comments for this pattern of logic

Reply:
@balamurugana 

one reason thant I put deleteObjectUploadsDir is that we must do it after call objectMPartPathLock.Unlock(), or it will fail

Reply:
A lock is required to do any operation else it would lead to unprotected operation.  Could you tell me what failure you see?

Reply:
@balamurugana 

Hi, you are right, it has nothing to do with objectMPartPathLock.Unlock()

Reply:
@fangyuxiangGL Yep. `rwlk.Close()` needs to be done prior to `objectMPartPathLock.Unlock()`.